onnxscript/rewriter/pattern.py

```diff
@@ -1511,6 +1512,7 @@ def rule(cls, *args, **kwargs):
         instance = cls(*args, **kwargs)
         setup = instance.setup if hasattr(instance, "setup") else None
         cleanup = instance.cleanup if hasattr(instance, "cleanup") else None
+        as_function = instance.as_function if hasattr(instance, "as_function") else False
```

Comment:
Is it possible to ensure all instances have the fields defined?

Reply:
Done